cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.exception;
+
+/**
+ * Class representing the category of an error.
+ *
+ * <p>This class is used to classify errors into different categories,
+ * such as plugin errors, provisioning errors, etc.
+ * </p>
+ */
+public class ErrorCategory {
+  public static final String PLUGIN = "Plugin";
+  public static final String OTHERS = "Others";
+  private final boolean isPlugin;
+  private final String errorCategory;
```

Comment:
When we say errorCategory don't we actually mean error sub category?  Looking at this variable it feels that it stores the whole category like "plugin - label" but it doesn't. Should we rename it or can we please add a comment to explain it?

Reply:
It is not necessarily error sub category, it does stores the whole category like `Plugin-'<STAGE_NAME>'` but if for some reason we don't have the stage name right now it just stores `Plugin`.